src/Symfony/Component/HttpKernel/EventListener/SessionListener.php

```diff
@@ -33,6 +34,13 @@ protected function getSession()
             return;
         }
 
+        if ($this->container->has('session_storage')
+            && ($storage = $this->container->get('session_storage')) instanceof NativeSessionStorage
+            && $this->container->get('request_stack')->getMasterRequest()->isSecure()
+        ) {
+            $storage->setOptions(array('cookie_secure' => true));
```

Comment:
Might be a silly question but are we sure that this is called before the storage `start`?

Reply:
We can throw a deprecation it that's not the case so that we ensure we can provide good security in 5.0. WDYT?

Reply:
Yep, sounds good to me 👍 

Also, this option should only be set when `framework.session.cookie_secure` === `auto`, not all the time. Looks like it is calling this option even if the option is set as `false` here.

Reply:
Actually, we don't care: `NativeSessionStorage::setOptions` starts with:
```php
if (headers_sent() || \PHP_SESSION_ACTIVE === session_status()) {
    return;
}
```

Reply:
But... we do care because if it is called after the `start` then the `cookie_secure` will not be configured which will end up having the secure cookie not configured 🤔